src/helpers/helpers.chart.js

```diff
@@ -36,8 +36,8 @@ function getChartDimensionByScale(scale, options) {
     };
   }
   return {
-    start: options.start,
-    end: options.end
+    start: Math.min(options.start, options.end),
```

Comment:
These are potentially scale values. How does it work with a category scale with arbitrary labels like `['10', '5', '20', '4']`?

Reply:
you're right, it doesn't work. Let me have a look.

Reply:
I was, wrong because I have checked and using the a category scale and it works (I'm preparing a test case):

```javascript
scales: {
  x: {
    type: 'category',
    display: true,
    labels: ['January', 'February', 'March', 'April', 'May', 'June', 'July'],
  },
  y: {
    display: false,
    min: -10,
    max: 50
  }
},
plugins: {
  annotation: {
    annotations: {
      first: {
        type: 'box',
        xMax: 'February',
        backgroundColor: 'rgba(159, 226, 191, 0.5)',
      },
      second: {
        type: 'box',
        xMax: 'April',
        xMin: 'February',
        backgroundColor: 'rgba(255, 191, 0, 0.5)',
      },
      third: {
        type: 'box',
        xMin: 'April',
        backgroundColor: 'rgba(255, 191, 0, 0.5)',
      }
    }
  }
}
```
![issueCoordinates1](https://user-images.githubusercontent.com/11741250/150786314-4738cabf-8599-4387-82d7-f25b3928a838.png)

Reply:
Do you have other use cases that I forgot or I'm not aware of?

Reply:
Yeah, actually the code changed here is in the branch where scale is not defined. So the added test does not go there.

Reply:
But I was wrong, those are not potentialy scale values, those are chartArea coordinates.

Reply:
> But I was wrong, those are not potentialy scale values, those are chartArea coordinates.

Yeah!

Reply:
> Yeah, actually the code changed here is in the branch where scale is not defined. So the added test does not go there.

Ok... do you mean PR #653 where x/yScale will not be anymore in the options?

Reply:
No, by branch I mean the `if` conditions in that function, specifically:
![image](https://user-images.githubusercontent.com/27971921/150810851-fb12029d-2f46-4d21-825d-1302cd19d76c.png)